cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/group/AbstractGcpGroupBuilder.java

```diff
@@ -16,4 +21,16 @@
         return checkResources(resourceType(), context, auth, resources);
     }
 
+    protected CloudResource doOperationalRequest(CloudResource resource, ComputeRequest<Operation> request) throws IOException {
+        try {
+            Operation operation = request.execute();
+            if (operation.getHttpErrorStatusCode() != null) {
+                throw new GcpResourceException(operation.getHttpErrorMessage(), resourceType(), resource.getName());
+            }
+            return createOperationAwareCloudResource(resource, operation);
+        } catch (GoogleJsonResponseException e) {
+            exceptionHandler(e, resource.getName(), resourceType());
+            throw new GcpResourceException(checkException(e), resourceType(), resource.getName());
```

Comment:
There's a case in `exceptionHandler` that does not thrown an exception, specifically the case where the resource was not found. Previously `GcpFirewallInResourceBuilder.delete` was only calling `exceptionHandler`, which means that in the case the resource wasn't found no exception was thrown, because you don't need to delete a resource that doesn't exist. With this change, that case will always through a `GcpResourceException`. I don't think that's a valid change because we're introduction an exception into a case that's actually not an error case.

Reply:
Yes, Changed is back to keeps deletes as a special case. and kept the pattern in the new builder